### PR TITLE
PhpUnitDedicateAssertFixer - fix to assertCount too

### LINKED ISSUE #3750: PHPUnit - fix to assertCount
A few years back there was an issue with `assertCount` when using an iterator on HHVM (https://github.com/sebastianbergmann/phpunit/pull/1413), as such we don't fix asserts in PHPUnit tests like;

```
$this->assertSame(1, count($a));
```

Since than;
- less projects support HHVM (we dropped it here as well) 
- HHVM and PHPUnit improved over time
- we have add a way to add a configuration option to fix this conditionally 

Therefor I suggest to add it.

### PR DESCRIPTION
**Pull request by SpacePossum**
closes #3750

```
            'assert same' => [
                '<?php $this->assertCount(1, $a);',
                '<?php $this->assertSame(1, count($a));',
            ],
```

for more example please see the tests

### CODE REVIEW

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Comment by TomasVotruba** [comment]
For more the rest of advanced cases I'd like to refer great job by @carusogabriel in Rector: https://github.com/rectorphp/rector/blob/master/docs/AllRectorsOverview.md#rectorphpunitrectorspecificmethodassertnotoperatorrector

```diff
-$this->assertTrue($foo === $bar, "message");
+$this->assertSame($bar, $foo, "message");

-$this->assertFalse($foo >= $bar, "message");
+$this->assertLessThanOrEqual($bar, $foo, "message");
```

```diff
-$this->assertFalse(strpos($anything, "foo"), "message");
+$this->assertNotContains("foo", $anything, "message");

-$this->assertNotFalse(stripos($anything, "foo"), "message");
+$this->assertContains("foo", $anything, "message");
```

```diff
-$this->assertTrue(property_exists(new Class, "property"), "message");
+$this->assertClassHasAttribute("property", "Class", "message");

-$this->assertFalse(property_exists(new Class, "property"), "message");
+$this->assertClassNotHasAttribute("property", "Class", "message");
```

etc.

File: tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php [hunk 1]
```diff
@@ -344,4 +344,136 @@ public function testInvalidConfig()
...
+        $this->doTest($expected, $input);
+    }
+
+    public function provideTestAssertCountCases()
+    {
+        return [
+            // positive fixing
+            'assert same' => [
+                '<?php $this->assertCount(1, $a);',
+                '<?php $this->assertSame(1, count($a));',
```
Comment (carusogabriel) [inline tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php]: What about `sizeof`? :thinking:

Comment (SpacePossum) [reply tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php]: should very easy to add, can someone make an issue/PR?

**Comment by SpacePossum** [comment]
nice list, maybe someone can backport to hear :)

File: tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php (hunk 1 above)
Comment (carusogabriel) [reply tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php]: @SpacePossum I'll :)

**Pull request merged**